### PR TITLE
refactor: use optional chaining when validating config rules

### PR DESCRIPTION
**Pull request by lucasrmendonca**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain: Minor refactoring

#### What changes did you make? (Give an overview)
This change makes the function more readable by removing unnecessary "&&" symbols

### CODE REVIEW

File: lib/config/flat-config-helpers.js [hunk 1]
```diff
@@ -65,13 +65,9 @@ function parseRuleId(ruleId) {
...
  */
 function getRuleFromConfig(ruleId, config) {
-
     const { pluginName, ruleName } = parseRuleId(ruleId);
 
-    const plugin = config.plugins && config.plugins[pluginName];
-    const rule = plugin && plugin.rules && plugin.rules[ruleName];
-
-    return rule;
+    return config?.plugins?.[pluginName]?.rules?.[ruleName];
```
Comment (fasttime) [inline lib/config/flat-config-helpers.js:70]: Thanks for the PR @lucasrmendonca. I seems that the `config` argument is not optional in this function. So maybe like this?

```suggestion
    return config.plugins?.[pluginName]?.rules?.[ruleName];
```

Comment (lucasrmendonca) [reply lib/config/flat-config-helpers.js]: Done!

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Leaving open for @fasttime to verify.

**Review by fasttime — APPROVED** [review]
LGTM, thanks!

**Pull request merged**